### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'as_stream'

### ISSUE THREAD

**Issue by epistoteles** [damnever/pigar#153]
**Describe the bug**
No matter in which project or environment I run pigar, `pigar generate` terminates with the following error:
```
AttributeError: 'NoneType' object has no attribute 'as_stream'
```

**Context (Environment)**
 - OS: Ubuntu 22.04
 - Version of `Python`: 3.10.6
 - Version of `pigar`: 2.0.6

**Additional context**
Full error:
```
/home/korbinian/.local/lib/python3.10/site-packages/pigar/_vendor/pip/_vendor/distlib/database.py:173: UserWarning: Unable to read distribution at /home/korbinian/.local/lib/python3.10/site-packages/lightning_lite-1.8.0.post1.dist-info, perhaps due to bad metadata: '.*' not allowed for '>=' constraints
  for dist in self._yield_distributions():
/home/korbinian/.local/lib/python3.10/site-packages/pigar/_vendor/pip/_vendor/distlib/database.py:173: UserWarning: Unable to read distribution at /home/korbinian/.local/lib/python3.10/site-packages/pytorch_lightning-1.8.0.post1.dist-info, perhaps due to bad metadata: '.*' not allowed for '>=' constraints
  for dist in self._yield_distributions():
/home/korbinian/.local/lib/python3.10/site-packages/pigar/_vendor/pip/_vendor/distlib/database.py:173: UserWarning: Unable to read distribution at /usr/lib/python3/dist-packages/Send2Trash-1.8.1b0.dist-info, perhaps due to bad metadata: invalid version: "darwin"
  for dist in self._yield_distributions():
Traceback (most recent call last):
  File "/home/korbinian/.local/bin/pigar", line 8, in <module>
    sys.exit(main())
  File "/home/korbinian/.local/lib/python3.10/site-packages/pigar/__main__.py", line 479, in main
    cli()
  File "/home/korbinian/.local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/korbinian/.local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/korbinian/.local/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/korbinian/.local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/korbinian/.local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/korbinian/.local/lib/python3.10/site-packages/pigar/__main__.py", line 239, in generate
    analyzer = RequirementsAnalyzer(project_path)
  File "/home/korbinian/.local/lib/python3.10/site-packages/pigar/core.py", line 36, in __init__
    self._installed_dists = installed_distributions()
  File "/home/korbinian/.local/lib/python3.10/site-packages/pigar/dist.py", line 212, in installed_distributions
    req = FrozenRequirement.from_dist(distribution)
  File "/home/korbinian/.local/lib/python3.10/site-packages/pigar/dist.py", line 122, in from_dist
    installed_files = [
  File "/home/korbinian/.local/lib/python3.10/site-packages/pigar/dist.py", line 122, in <listcomp>
    installed_files = [
  File "/home/korbinian/.local/lib/python3.10/site-packages/pigar/_vendor/pip/_vendor/distlib/database.py", line 677, in list_installed_files
    for result in self._get_records():
  File "/home/korbinian/.local/lib/python3.10/site-packages/pigar/_vendor/pip/_vendor/distlib/database.py", line 596, in _get_records
    with contextlib.closing(r.as_stream()) as stream:
AttributeError: 'NoneType' object has no attribute 'as_stream'
```
 
When using `pigar -l DEBUG`, the last line printed before the error starts is
```
found local distribution: <PyNaCl 1.5.0  [nacl _sodium]  [/usr/lib/python3/dist-packages/nacl]>
```
(after a lot of other packages)

**Comment by damnever** [comment]
What is the output of `pip freeze`?

**Comment by epistoteles** [comment]
This is the output of pip freeze:

```
absl-py==1.3.0
aif360==0.5.0
aiohttp==3.8.3
aiosignal==1.2.0
appdirs==1.4.4
apturl==0.5.2
argon2-cffi==21.1.0
arrow==1.2.3
astunparse==1.6.3
async-timeout==4.0.2
attrs==22.1.0
Babel==2.8.0
backcall==0.2.0
bcrypt==3.2.0
beautifulsoup4==4.10.0
beniget==0.4.1
black==23.3.0
bleach==4.1.0
blinker==1.4
Brlapi==0.8.3
Brotli==1.0.9
cachetools==5.2.0
caffeine==2.9.8
cairocffi==1.4.0
CairoSVG==2.7.0
certifi==2020.6.20
cffi==1.15.1
cfgv==3.3.1
chardet==4.0.0
charset-normalizer==2.1.1
chex==0.1.5
click==8.1.3
cloudpickle==2.2.0
codecarbon==2.1.4
colorama==0.4.4
command-not-found==0.3
contourpy==1.0.6
cryptography==3.4.8
cssselect2==0.7.0
cupshelpers==1.0
cycler==0.11.0
dbus-python==1.2.18
decorator==4.4.2
defer==1.0.6
defusedxml==0.7.1
diffusers==0.11.1
dill==0.3.6
distlib==0.3.6
distro==1.7.0
distro-info===1.1build1
dm-haiku==0.0.8
dm-tree==0.1.7
docker-pycreds==0.4.0
docopt==0.6.2
duplicity==0.8.21
entrypoints==0.4
etils==0.9.0
evdev==1.6.1
ewmh==0.1.6
exceptiongroup==1.0.0
fasteners==0.14.1
fastjsonschema==2.16.3
filelock==3.8.2
filetype==1.2.0
fire==0.4.0
flatbuffers==22.10.26
fonttools==4.38.0
frozenlist==1.3.1
fs==2.4.12
fsspec==2022.10.0
future==0.18.2
fuzzywuzzy==0.18.0
gast==0.4.0
gitdb==4.0.9
GitPython==3.1.29
google-auth==2.17.1
google-auth-oauthlib==1.0.0
google-pasta==0.2.0
googleapis-common-protos==1.56.4
graphviz==0.20.1
grpcio==1.50.0
gyp==0.1
h5py==3.7.0
html5lib==1.1
httplib2==0.20.2
huggingface-hub==0.11.1
identify==2.5.22
idna==3.3
imbalanced-learn==0.9.1
importlib-metadata==4.6.4
importlib-resources==5.10.0
iniconfig==1.1.1
ipykernel==6.7.0
ipython==7.31.1
ipython_genutils==0.2.0
ipywidgets==6.0.0
jax==0.3.24
jaxlib==0.3.24
jedi==0.18.0
jeepney==0.7.1
Jinja2==3.0.3
jmp==0.0.2
joblib==1.2.0
jsonschema==3.2.0
jupyter-client==7.1.2
jupyter-core==4.9.1
jupyterlab-pygments==0.1.2
keras==2.12.0
Keras-Preprocessing==1.1.2
keyring==23.5.0
kiwisolver==1.4.4
language-selector==0.1
launchpadlib==1.10.16
lazr.restfulclient==0.14.4
lazr.uri==1.0.6
libclang==14.0.6
lightning-lite==1.8.0.post1
lightning-utilities==0.3.0
llvmlite==0.39.1
lockfile==0.12.2
louis==3.20.0
lxml==4.8.0
lz4==3.1.3+dfsg
macaroonbakery==1.3.1
Mako==1.1.3
Markdown==3.4.1
MarkupSafe==2.1.1
matplotlib==3.4.0
matplotlib-inline==0.1.3
memory-profiler==0.61.0
monotonic==1.6
more-itertools==8.10.0
mpmath==0.0.0
multidict==6.0.2
mypy-extensions==1.0.0
nbclient==0.5.6
nbconvert==6.4.0
nbformat==5.8.0
nest-asyncio==1.5.4
netifaces==0.11.0
nodeenv==1.7.0
notebook==6.4.8
notify2==0.3
numba==0.56.4
numpy==1.23.4
nvidia-cublas-cu11==11.10.3.66
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cudnn-cu11==8.5.0.96
oauthlib==3.2.0
olefile==0.46
openai==0.27.4
opt-einsum==3.3.0
optax==0.1.3
packaging==23.0
pandas==1.5.1
pandocfilters==1.5.0
paramiko==2.9.3
parso==0.8.1
pathspec==0.11.1
pathtools==0.1.2
pexpect==4.8.0
pickleshare==0.7.5
pigar==2.0.6
Pillow==9.4.0
pipdeptree==2.7.0
pipreqs==0.4.13
pipreqsnb==0.2.4
platformdirs==3.2.0
plotly==5.11.0
pluggy==1.0.0
ply==3.11
pre-commit==3.2.2
prometheus-client==0.9.0
promise==2.3
prompt-toolkit==3.0.28
protobuf==3.20.3
psutil==5.9.3
ptyprocess==0.7.0
py==1.10.0
py-cpuinfo==9.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycairo==1.20.1
pycparser==2.21
pycups==2.0.1
Pygments==2.11.2
PyGObject==3.42.1
PyJWT==2.3.0
pymacaroons==0.13.0
PyNaCl==1.5.0
pynput==1.7.6
pynvml==11.5.0
pyparsing==2.4.7
PyQt5==5.15.6
PyQt5-sip==12.9.1
pyRFC3339==1.1
pyrsistent==0.18.1
PySide6==6.4.3
PySide6-Addons==6.4.3
PySide6-Essentials==6.4.3
pytest==7.2.0
python-apt==2.4.0+ubuntu1
python-dateutil==2.8.1
python-debian===0.1.43ubuntu1
python-xlib==0.33
pythran==0.10.0
pytorch-lightning==1.8.0.post1
PyTurboJPEG==1.7.0
pytz==2022.1
pyxdg==0.27
PyYAML==5.4.1
pyzmq==22.3.0
Random-Word==1.0.11
rebasin==0.0.22
regex==2022.10.31
reportlab==3.6.8
requests==2.25.1
requests-oauthlib==1.3.1
rsa==4.9
scikit-learn==1.1.3
scipy==1.9.3
scour==0.38.2
seaborn==0.12.1
SecretStorage==3.3.1
Send2Trash==1.8.1b0
sentry-sdk==1.10.1
setproctitle==1.3.2
setuptools-scm==7.0.5
shap==0.41.0
shiboken6==6.4.3
shortuuid==1.0.1
six==1.16.0
sklearn==0.0.post1
slicer==0.0.7
smmap==5.0.0
soupsieve==2.3.1
streamdeck==0.9.3
streamdeck-ui==2.0.15
sympy==1.9
systemd-python==234
tableprint==0.9.1
tabulate==0.9.0
tempeh==0.1.12
tenacity==8.1.0
tensorboard==2.12.1
tensorboard-data-server==0.7.0
tensorboard-plugin-wit==1.8.1
tensorflow==2.12.0
tensorflow-datasets==4.7.0
tensorflow-estimator==2.12.0
tensorflow-io-gcs-filesystem==0.27.0
tensorflow-metadata==1.10.0
termcolor==2.1.0
terminado==0.13.1
testpath==0.5.0
threadpoolctl==3.1.0
tinycss2==1.2.1
toml==0.10.2
tomli==2.0.1
toolz==0.12.0
torch==1.13.0
torchmetrics==0.10.2
torchvision==0.14.0
tornado==6.1
tqdm==4.64.1
traitlets==5.1.1
typing_extensions==4.4.0
ubuntu-advantage-tools==8001
ubuntu-drivers-common==0.0.0
ufoLib2==0.13.1
ufw==0.36.1
unattended-upgrades==0.1
unicodedata2==14.0.0
urllib3==1.26.12
usb-creator==0.3.7
virtualenv==20.21.0
wadllib==1.3.6
wandb==0.13.4
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==2.2.2
widgetsnbextension==2.0.0
wrapt==1.14.1
xdg==5
xkit==0.0.0
yarg==0.1.9
yarl==1.8.1
zipp==1.0.0
```

**Comment by pipari** [comment]
I also have this issue.

anyio==3.6.2
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
attrs==23.1.0
autograd==1.5
autograd-gamma==0.5.0
Babel==2.12.1
backcall==0.2.0
beautifulsoup4==4.12.2
bleach==6.0.0
Brotli==1.0.9
certifi==2023.5.7
cffi==1.15.1
click==8.1.3
click-plugins==1.1.1
cligj==0.7.2
colorama==0.4.6
comm==0.1.3
contourpy==1.0.7
cssselect2==0.7.0
cycler==0.11.0
debugpy==1.6.7
decorator==5.1.1
defusedxml==0.7.1
docutils==0.17.1
docxcompose==1.4.0
entrypoints==0.4
et-xmlfile==1.1.0
executing==1.2.0
fastjsonschema==2.17.1
Fiona==1.9.4.post1
fonttools==4.39.4
fqdn==1.5.1
future==0.18.3
geopandas==0.12.2
greenlet==2.0.2
h3==3.7.6
html5lib==1.1
idna==3.4
imageio==2.29.0
ipyevents==2.0.1
ipykernel==6.20.2
ipympl==0.9.3
ipython==8.8.0
ipython-genutils==0.2.0
ipywidgets==8.0.6
isoduration==20.11.0
jedi==0.18.2
Jinja2==3.1.2
joblib==1.2.0
jsonpointer==2.3
jsonschema==4.17.3
jupyter-console==6.4.4
jupyter-events==0.6.3
jupyter_client==7.4.9
jupyter_core==5.1.5
jupyter_server==2.5.0
jupyter_server_terminals==0.4.4
jupyterlab-pygments==0.2.2
jupyterlab-widgets==3.0.7
kaleido==0.2.1
kiwisolver==1.4.4
llvmlite==0.39.1
lxml==4.9.2
MarkupSafe==2.1.2
matplotlib==3.6.3
matplotlib-inline==0.1.6
mistune==2.0.5
mplcursors==0.5.2
nbclassic==1.0.0
nbclient==0.7.4
nbconvert==7.2.10
nbformat==5.7.3
nest-asyncio==1.5.6
networkx==3.1
notebook==6.5.4
notebook_shim==0.2.3
numba==0.56.4
numpy==1.23.5
openpyxl==3.0.10
packaging==23.1
pandas==1.5.3
pandocfilters==1.5.0
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.5.0
platformdirs==3.5.1
plotly==5.13.1
prometheus-client==0.16.0
prompt-toolkit==3.0.38
psutil==5.9.5
ptyprocess==0.7.0
pure-eval==0.2.2
pybaselines==1.0.0
pycparser==2.21
pydyf==0.6.0
Pygments==2.15.1
pyparsing==3.0.9
pyphen==0.14.0
pyproj==3.4.1
pyrsistent==0.19.3
python-dateutil==2.8.2
python-docx==0.8.11
python-dotenv==0.21.1
python-json-logger==2.0.7
pytz==2023.3
PyWavelets==1.4.1
PyYAML==6.0
pyzmq==25.0.2
qtconsole==5.4.3
QtPy==2.3.1
reliability==0.8.7
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
scikit-gstat==1.0.10
scikit-image==0.19.3
scikit-learn==1.2.2
scipy==1.10.1
seaborn==0.12.2
Send2Trash==1.8.2
shapely==2.0.1
six==1.16.0
sniffio==1.3.0
soupsieve==2.4.1
SQLAlchemy==1.4.48
stack-data==0.6.2
tenacity==8.2.2
terminado==0.17.1
threadpoolctl==3.1.0
tifffile==2023.4.12
timezonefinder==6.1.1
tinycss2==1.2.1
tornado==6.3.2
tqdm==4.65.0
traitlets==5.9.0
uri-template==1.2.0
wcwidth==0.2.6
weasyprint==57.2
webcolors==1.13
webencodings==0.5.1
websocket-client==1.5.2
widgetsnbextension==4.0.7
windrose==1.8.1
zopfli==0.2.2

**Comment by damnever** [comment]
I am unable to reproduce this issue.

Please try this to see if the error persists:
```
pip uninstall pigar
pip install git+https://github.com/damnever/pigar.git@fix/153
```

**Comment by thiswillbeyourgithub** [comment]
> I am unable to reproduce this issue.
> 
> Please try this to see if the error persists:
> 
> ```
> pip uninstall pigar
> pip install git+https://github.com/damnever/pigar.git@fix/153
> ```


Installed pigar today (thanks!): same issue, fixed by your link

**Comment by damnever** [comment]
A new release is now available, give it a try: `pip install pigar==2.0.7`.
